tests/unit/test_errorhandler.py

```diff
@@ -10,101 +10,74 @@
-        else:
-            self.fail("Expected errorhandler.ClientError to be raised "
-                      "but no exception was raised.")
-
-    def test_error_handler_server_side(self):
-        response = {
-            'Error': {'Code': 'InternalError',
-                      'HostId': 'foohost',
-                      'Message': 'An internal error has occurred',
-                      'RequestId': 'requestid'},
-            'ResponseMetadata': {}}
-        handler = errorhandler.ErrorHandler()
-        http_response = self.create_http_response(status_code=500)
-        # We're manually using the try/except form because
-        # we want to catch the exception and assert that it has specific
-        # attributes on it.
-        operation = mock.Mock()
-        operation.name = 'OperationName'
-        try:
-            handler(http_response, response, operation)
-        except errorhandler.ServerError as e:
-            # First, the operation name should be in the error message.
-            self.assertIn('OperationName', str(e))
-            # We should state that this is a ServerError.
-            self.assertIn('server error', str(e))
-            # And these values should be available on the exception
-            # so clients can access this information programmatically.
-            self.assertEqual(e.error_code, 'InternalError')
-            self.assertEqual(e.error_message, 'An internal error has occurred')
-            self.assertEqual(e.operation_name, 'OperationName')
-        except Exception as e:
-            self.fail("Unexpected error raised: %s" % e)
-        else:
-            self.fail("Expected errorhandler.ServerError to be raised "
-                      "but no exception was raised.")
-
-    def test_no_exception_raised_on_200(self):
-        response = {
-            'CommonPrefixes': [],
-            'Contents': [],
-        }
-        handler = errorhandler.ErrorHandler()
-        http_response = self.create_http_response(status_code=200)
-        # We're manually using the try/except form because
-        # we want to catch the exception and assert that it has specific
-        # attributes on it.
-        operation = mock.Mock()
-        operation.name = 'OperationName'
-        try:
-            self.assertIsNone(handler(http_response, response, operation))
-        except errorhandler.BaseOperationError as e:
-            self.fail("Unexpected error raised: %s" % e)
-
+            cr = self.error_handlers_chain.handle_exception(e, stdout, stderr)
+            self.assertEqual(cr, case.rc, case.exception.__class__)
+            self.assertIn(case.message, stderr.getvalue())
+            self.assertEqual('', stdout.getvalue())
 
-if __name__ == '__main__':
-    unittest.main()
+    def test_error_handling(self):
+        Case = namedtuple('Case', [
+            'exception',
+            'rc',
+            'message',
+        ])
+        cases = [
+            Case(Exception(), 255, '\n\n'),
```

Comment:
Instead of asserting the newlines, let's adding a message to the exception and assert that the exception message is in the stderr. So to update this example:
```
Case(Exception('exception msg'), 255, 'exception msg')
```
I like this more because it is not really clear for any of the assertions that only use newlines or even empty strings why we would be asserting it. By asserting that the error message is in stderr is a bit more clear.